explainer.md

```diff
@@ -74,6 +74,147 @@ const cameraTexture = binding.getCameraImage(view.camera);
+
+**Note:** The steps to obtain camera intrinsics and extrinsics are only valid since the camera image returned by the API is perfectly aligned with the `XRView` from which it was obtained.
+
+### Camera space to screen space - projection matrix and viewport route
+
+Projection matrix convetions as per: http://www.songho.ca/opengl/gl_projectionmatrix.html.
+
+Starting from a projection matrix with the following parameters:
+
+```
+P = p0  p4  p8   p12
+    p1  p5  p9   p13
+    p2  p6  p10  p14
+    p3  p7  p11  p15
+```
+
+We can repeat the derivation of the projection matrix from [songho.ca](http://www.songho.ca/opengl/gl_projectionmatrix.html), allowing for non-zero skew:
+
+```
+P = p0  p4  p8   0    = 2n/(r-l)  skew      (r+l)/(r-l)  0
+    0   p5  p9   0      0         2n/(t-b)  (t+b)/(t-b)  0
+    0   0   p10  p14    0         0        -(f+n)/(f-n) -2fn/(f-n)
+    0   0  -1    0      0         0        -1            0
+```
+
+The skew factor controls how much of the Y coordinate is mixed into the X coordinate. It is usually zero, but WebXR allows nonzero skew values which results in nonrectangular pixels.
+
+A GL projection matrix transforms from camera space to clip space, then to NDC after perspective divide. This needs to be scaled to pixels based on the viewport `vp`. The NDC x and y ranges `(-1 .. 1)` are then transformed to `(vp.x .. vp.x + vp.width)` and `(vp.y .. vp.y + vp.height)`, respectively. For example, NDC x coordinate is transformed to screen space:
+
+```
+screen_x = vp.w * (ndc_x + 1)/2 + vp.x
+         = (vp.w/2) * ndc_x + (vp.w/2 + vp.x)
+```
+
+Using a matrix S for the NDC-to-screen-coordinate transform described above, this becomes:
+
+```
+p_screen.xy = (S * p_ndc).xy
+
+with S = vp.w/2  0       0  vp.w/2 + vp.x
+         0       vp.h/2  0  vp.h/2 + vp.y
+         0       0       1  0
+         0       0       0  1
+```
+
+The camera-space point transformation into screen space is then as follows:
+
+```
+p_screen.xy = (S * p_ndc).xy
+            = (S * p_clip).xy / p_clip.w
+            = (S * P * p_camera).xy / (P * p_camera).w
+            = (S * P * p_camera).xy / (-p_camera.z)
+```
+
+Note that this uses the usual GL convention of looking along the negative Z axis, with negative-z points being visible.
+
+### Camera space to screen space - intrinsic matrix route
+
+Intrinsic matrix convention as per https://en.wikipedia.org/wiki/Camera_resectioning#Intrinsic_parameters
+
+The intrinsic matrix K transforms from camera space to homogenous screen space, providing pixel screen coordinates after the perspective divide. This convention assumes looking along the positive Z axis, with positive-z points being visible.
+
+```
+K = ax  gamma  u0  0
+    0   ay     v0  0
+    0   0      1   0
+```
+
+For compatibility with WebXR, insert a placeholder 3rd row to get a 4x4 matrix `Kexp` and invert the Z coordinate. This produces a modified intrinsic matrix K':
```

Comment:
I think the "and" is a bit confusing here. How about:

> For compatibility with WebXR, create a 4x4 matrix `Kexp` by inserting a placeholder 3rd row. (The values in this row don't affect the intrinsic parameter calculation and are marked by `*` .) Then invert the Z coordinate to produce a modified intrinsic matrix K':

Reply:
Done.